node/impl/full/eth.go

```diff
@@ -976,6 +977,46 @@ func (a *EthModule) EthTraceReplayBlockTransactions(ctx context.Context, blkNum
 	return allTraces, nil
 }
 
+func (a *EthModule) EthTraceTransaction(ctx context.Context, txHash string) ([]*ethtypes.EthTraceTransaction, error) {
+
+	// convert from string to internal type
+	ethTxHash, err := ethtypes.ParseEthHash(txHash)
+	if err != nil {
+		return nil, xerrors.Errorf("cannot parse eth hash: %w", err)
+	}
+
+	tx, err := a.EthGetTransactionByHash(ctx, &ethTxHash)
+	if err != nil {
+		return nil, xerrors.Errorf("cannot get transaction by hash: %w", err)
+	}
+
+	if tx == nil {
+		return nil, xerrors.Errorf("transaction not found")
+	}
+
+	blockTraces, err := a.EthTraceBlock(ctx, strconv.FormatUint(uint64(*tx.BlockNumber), 10))
```

Comment:
The `BlockNumber` will be nil if the transaction is still in the mpool/pending. We should just error out in this case as there's no trace for pending transactions and document the API to explicitly say so.

Reply:
changed here a46b3d71a